### PR TITLE
Support adding/porting to ES6 components - Port Button-related things to ES6 to start

### PR DESCRIPTION
**Pull request by jwietelmann**
Also deprecating Rev.Mixins.Core.getNormalizedChildren. When it was written, either React.Children.map did not exist, or we just didn't know about it.

### CODE REVIEW

File: app/assets/javascripts/revelry.js [hunk 1]
```diff
@@ -5,5 +5,6 @@
 //= require revelry/Main
 //= require revelry/router/router
 //= require revelry/router/routerlink
+//= require revelry/ui/buttons/Button
```
Comment (jwietelmann) [inline app/assets/javascripts/revelry.js]: Was just using this to see whether ES6 files were even being recognized. They aren't. I'll kill this off when I hunt down the problem.

File: app/assets/javascripts/revelry/ui/buttons/Button.es6 [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    console.log(this.classAdd)
+    console.log(this.getPropsWithout)
+    return <button {...this.passThroughProps} className={this.className}>
+      {this.props.children}
+    </button>
+  }
+
+}
+
+_.extend(Button.prototype, Rev.Mixins.Core)
```
Comment (prehnRA) [inline app/assets/javascripts/revelry/ui/buttons/Button.es6]: You know that if we require the extensions/mixins thing I wrote you would get this for free _and_ mixins would work with ES6 components, right?

File: app/assets/javascripts/revelry/ui/buttons/Button.es6 [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    })
+    return this.classAdd(classNamesObject)
+  }
+
+  get passThroughProps() {
+    return this.getPropsWithout.call(this._propsForClassNames)
+  }
+
+  render() {
+    console.log(this.classAdd)
```
Comment (prehnRA) [inline app/assets/javascripts/revelry/ui/buttons/Button.es6]: CRUSH IT

File: app/assets/javascripts/revelry/ui/buttons/Button.es6 (hunk 2 above)
Comment (jwietelmann) [inline app/assets/javascripts/revelry/ui/buttons/Button.es6]: Yes, I do... What I didn't know was that it wasn't required by default and that I should look in `extensions`. So: Require by default? Or require on a per-file basis?

**Pull request merged**